util/dockernode.go

```diff
@@ -39,22 +39,33 @@ func (dn *DockerNode) Start(args []string) error {
 	}
 
 	id := bytes.TrimSpace(out)
+	dn.ID = string(id)
 	idfile := filepath.Join(dn.Dir, "dockerID")
 	err = ioutil.WriteFile(idfile, id, 0664)
+
 	if err != nil {
+    	killContainer(dn.ID)
 		return err
 	}
 
-	dn.ID = string(id)
-
 	err = waitOnAPI(dn)
 	if err != nil {
+    	killContainer(dn.ID)
 		return err
 	}
 
 	return nil
 }
 
+func killContainer(id string) {
```

Comment:
Do you think we would be able to make use of the existing [`Kill()`](https://github.com/ipfs/iptb/pull/53/files#diff-b7342e471c774078f7baddd77adb6051R119) method here instead?

Reply:
Right, I think that's a good idea. I do have a question on proper error handling in Go though -- the `Kill()` method my return an error, but when we call it in `Start()` we already have an error to return. Is it best to print the error returned by `Kill()` and then return the one we had in `Start()`? Or concatenate them into one big error that we then return? In the code I have here I opted for the former approach, but wasn't certain.

Reply:
I would concatenate the error like you said, there is a `Wrap` method in the [golang errors package.](https://github.com/pkg/errors/blob/master/errors.go#L180)
We have gx'd the package here [QmVmDhyTTUcQXFD1rRQ64fGLMSAoaQvNH3hwuaCFAPq2hy](https://ipfs.io/ipfs/QmVmDhyTTUcQXFD1rRQ64fGLMSAoaQvNH3hwuaCFAPq2hy/errors)

Reply:
Nice, I'll use that, thanks!

Reply:
After looking at `Wrap()` and a few examples, it seemed a bit different than what I was looking for. Ended up using `fmt.Errorf()` instead.